Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvMeasurementWriter.hpp

```diff
@@ -0,0 +1,95 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/EventData/Measurement.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/GeometryHierarchyMap.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "ActsExamples/Digitization/SmearingAlgorithm.hpp"
+#include "ActsExamples/EventData/Cluster.hpp"
+#include "ActsExamples/EventData/Index.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/Framework/WriterT.hpp"
+
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include <TTree.h>
+
+class TFile;
+class TTree;
+
+namespace ActsExamples {
+
+/// @class CsvMeasurementWriter
+///
+/// This writes multiples file per event containing information about the
+/// measurement, the associated truth information and the cell/channel details
+///
+///     event000000001-cells.csv
+///     event000000001-measurements.csv
+///     event000000001-truth.csv
+///     event000000002-cells.csv
+///     event000000002-measurements.csv
+///     event000000002-truth.csv
+///     ...
+///
+/// Intrinsically thread-safe as one file per event
+class CsvMeasurementWriter final : public WriterT<MeasurementContainer> {
+ public:
+  struct Config {
+    /// Which measurement collection to write.
+    std::string inputMeasurements;
+    /// Which cluster collection to write (optional)
+    std::string inputClusters;
+    /// Which simulated (truth) hits collection to use.
+    std::string inputSimHits;
+    /// Input collection to map measured hits to simulated hits.
+    std::string inputMeasurementSimHitsMap;
+    /// Where to place output files
+    std::string outputDir;
+    /// Write the optional measurement surfaces collection
+    bool writeMeasurementSurfaces = false;
```

Comment:
Is this `writeMeasurementSurfaces` playing a role somewhere?

Reply:
I guess we are not writing out the  `MeasurementParticlesMap` because it could be derived from the `MeasurementSimHitsMap`, right?